WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+                XS_2016=0.1194, XSSource_2016="XSDB",
+                XS_2017=0.1313, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=0.1315, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.627, kFactorSource_2016="XSDB NNLO/LO=831.76/511.3",
+                kFactor_2017=1.679, kFactorSource_2017="XSDB/GenXSecAnalyzer NNLO/LO=831.76/495.3",
+                kFactor_2018=1.693, kFactorSource_2018="XSDB/GenXSecAnalyzer NNLO/LO=831.76/491.4",
+            ),
         },
         "TTJets_HT-2500toInf" : {
-            "CrossSection" : XSValues(XS_13TeV=0.002351015),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.627),
+            "CrossSection" : XSValues(
+                XS_2016=0.001445, XSSource_2016="XSDB",
+                XS_2017=0.001410, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=0.001420, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.627, kFactorSource_2016="XSDB NNLO/LO=831.76/511.3",
+                kFactor_2017=1.679, kFactorSource_2017="XSDB/GenXSecAnalyzer NNLO/LO=831.76/495.3",
+                kFactor_2018=1.693, kFactorSource_2018="XSDB/GenXSecAnalyzer NNLO/LO=831.76/491.4",
+            ),
         },
         "TTTo2L2Nu" : {
-            "CrossSection" : XSValues(XS_13TeV=88.29),
+            "CrossSection" : XSValues(
+                XS_13TeV=88.29, XSSource_13TeV="XSDB (NNLO)",
+            ),
         },
         "TTToHadronic" : {
-            "CrossSection" : XSValues(XS_13TeV=377.96),
+            "CrossSection" : XSValues(
+                XS_13TeV=377.96, XSSource_13TeV="XSDB (NNLO)",
+            ),
         },
         "TTToSemiLeptonic" : {
-            "CrossSection" : XSValues(XS_13TeV=365.34),
+            "CrossSection" : XSValues(
+                XS_13TeV=365.34, XSSource_13TeV="XSDB (NNLO)",
+            ),
         },
         "TTGamma_SingleLeptFromT" : {
-            "CrossSection" : XSValues(XS_13TeV=0.704),
+            "CrossSection" : XSValues(
+                XS_2016=0.7684, XSSource_2016="XSDB (NLO)",
+                XS_2017=0.7040, XSSource_2017="XSDB (LO)",
+                XS_2018=0.7053, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "TTGamma_SingleLeptFromTbar" : {
-            "CrossSection" : XSValues(XS_13TeV=0.704),
+            "CrossSection" : XSValues(
+                XS_2016=0.7659, XSSource_2016="XSDB (NLO)",
+                XS_2017=0.7040, XSSource_2017="XSDB (LO)",
+                XS_2018=0.7028, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "TTGamma_Dilept" : {
-            "CrossSection" : XSValues(XS_13TeV=0.5804),
+            "CrossSection" : XSValues(
+                XS_2016=0.6343, XSSource_2016="XSDB (NLO)",
+                XS_2017=0.5804, XSSource_2017="XSDB (LO)",
+                XS_2018=1.495,  XSSource_2018="GenXSecAnalyzer",
```

Comment:
do we understand why this is x3 from 2017?

Reply:
We don't understand this one yet. I can tell you that the Madgraph proc definitions are different, but I'm not really sure what effect the differences would have, if any. For reference, here are the relevant parts of the 2017 and 2018 process/madevent/Cards/proc_card_mg5.dat files:

2017:
```
import model sm-ckm_no_b_mass
define p = g u c d s u~ c~ d~ s~
define j = g u c d s u~ c~ d~ s~
define l+ = e+ mu+
define l- = e- mu-
define vl = ve vm vt
define vl~ = ve~ vm~ vt~
define p = 21 2 4 1 3 -2 -4 -1 -3 5 -5 # pass to 5 flavors
define j = p
define p = p b b~
define j = j b b~
define part =  ve vm vt e- mu- ve~ vm~ vt~ e+ mu+ ta- ta+
generate p p > t t~ a, t > b part part, t~ > b~ part part  @1
add process p p > t t~, t > b part part a, t~ > b~ part part  @2
add process p p > t t~, t > b part part , t~ > b~ part part a  @3
```

2018:
```
import model sm-no_b_mass
define p = g u c d s u~ c~ d~ s~
define j = g u c d s u~ c~ d~ s~
define l+ = e+ mu+
define l- = e- mu-
define p = 21 2 4 1 3 -2 -4 -1 -3 5 -5 # pass to 5 flavors
define j = p
define p = g u c d s b u~ c~ d~ s~ b~
define j = g u c d s b u~ c~ d~ s~ b~
define l+ = e+ mu+ ta+
define l- = e- mu- ta-
define vl = ve vm vt
define vl~ = ve~ vm~ vt~
generate p p > t t~ > l+ vl l- vl~ b  b~ a
```

Reply:
We've found the issue here. The following samples have different phase space cuts (notice the "_" vs "-"):
```
/TTGamma_Dilept_TuneCP5_13TeV-madgraph-pythia8/RunIIAutumn18MiniAOD-102X_upgrade2018_realistic_v15-v1/MINIAODSIM
/TTGamma_Dilept_TuneCP5_13TeV_madgraph_pythia8/RunIIAutumn18MiniAOD-102X_upgrade2018_realistic_v15-v1/MINIAODSIM
```
The sample using the "-" has a cross section of 1.495, while the the one with the "_" has a cross section of 0.5799. TreeMaker uses the samples with "_" for all three years, so this is the value taken.